tests/cache/models/samples/updateSampleAttributes.js

```diff
@@ -0,0 +1,346 @@
+      relatedLinks: '[]',
+      statusChangedAt: '2018-10-29T22:42:30.938Z',
+      provider: '84f3560b-1b68-4868-bea3-8dd0ad8aa8f3',
+      updatedAt: '2018-10-29T22:42:30.938Z',
+    };
+    const asp = {
+      id: '50513365-f24f-455c-8d47-c507d1c62a96',
+      isDeleted: '0',
+      relatedLinks: [],
+      tags: [],
+      description: 'this is a0 description',
+      imageUrl: 'http://www.bar.com/a0.jpg',
+      isPublished: 'true',
+      name: '___TEST_ASPECT',
+      timeout: '30s',
+      valueLabel: 's',
+      valueType: 'NUMERIC',
+      criticalRange: [0, 1],
+      warningRange: [2, 3],
+      infoRange: [4, 5],
+      okRange: [6, 7],
+      updatedAt: '2018-10-29T22:42:30.918Z',
+      createdAt: '2018-10-29T22:42:30.918Z',
+      writers: [],
+    };
+    updateSampleAttributes(qb, samp, asp);
+    expect(qb).to.deep.equal({
+      name: '___TEST_SUBJECT|___TEST_ASPECT',
+      value: '1',
+      subjectId: 'd2dfc000-5498-4bf0-a8c4-9c42c4569f05',
+      aspectId: '50513365-f24f-455c-8d47-c507d1c62a96',
+      previousStatus: 'Invalid',
+      statusChangedAt: '2018-10-29T22:42:30.938Z',
+      status: 'Critical',
+      updatedAt: '2018-10-30T20:24:37.053Z',
+    });
+  });
+
+  it('value is undefined', () => {
+    const qb = {
+      name: '___TEST_SUBJECT|___TEST_ASPECT',
+      value: undefined,
+      subjectId: 'd2dfc000-5498-4bf0-a8c4-9c42c4569f05',
+      aspectId: '50513365-f24f-455c-8d47-c507d1c62a96',
+    };
+    const samp = {
+      status: 'Critical',
+      value: '1',
+      previousStatus: 'Invalid',
+      user: '{"name":"___testUser@refocus.com",' +
+        '"email":"___testUser@refocus.com",' +
+        '"profile":{"name":"___testProfile"}}',
+      createdAt: '2018-10-29T22:42:30.938Z',
+      name: '___TEST_SUBJECT|___TEST_ASPECT',
+      subjectId: 'd2dfc000-5498-4bf0-a8c4-9c42c4569f05',
+      aspectId: '50513365-f24f-455c-8d47-c507d1c62a96',
+      relatedLinks: '[]',
+      statusChangedAt: '2018-10-29T22:42:30.938Z',
+      provider: '84f3560b-1b68-4868-bea3-8dd0ad8aa8f3',
+      updatedAt: '2018-10-29T22:42:30.938Z',
+    };
+    const asp = {
+      id: '50513365-f24f-455c-8d47-c507d1c62a96',
+      isDeleted: '0',
+      relatedLinks: [],
+      tags: [],
+      description: 'this is a0 description',
+      imageUrl: 'http://www.bar.com/a0.jpg',
+      isPublished: 'true',
+      name: '___TEST_ASPECT',
+      timeout: '30s',
+      valueLabel: 's',
+      valueType: 'NUMERIC',
+      criticalRange: [0, 1],
+      warningRange: [2, 3],
+      infoRange: [4, 5],
+      okRange: [6, 7],
+      updatedAt: '2018-10-29T22:42:30.918Z',
+      createdAt: '2018-10-29T22:42:30.918Z',
+      writers: [],
+    };
+    updateSampleAttributes(qb, samp, asp);
+    expect(qb).to.deep.equal({
+      name: '___TEST_SUBJECT|___TEST_ASPECT',
+      value: '',
+      subjectId: 'd2dfc000-5498-4bf0-a8c4-9c42c4569f05',
+      aspectId: '50513365-f24f-455c-8d47-c507d1c62a96',
+      previousStatus: 'Critical',
```

Comment:
Why isn't this invalid?

Reply:
In this case, the user passed in a "value" attribute, but the value itself (undefined) does not map to a status, so that means the new status is "Invalid" and the previousStatus becomes the "status" from the previous version of the this sample (samp), which was "Critical".